airgun/widgets.py

```diff
@@ -1718,3 +1719,35 @@ def fill(self, values):
                 widget = get_widget_by_name(self, key)
                 if widget:
                     widget.fill(value)
+
+
+class ToggleButton(Button):
+    """A simple toggle button that we can read/write it's state via the standard view functions
+    read/fill
+    """
+
+    def __init__(self, parent, *text, locator=None, **kwargs):
+        self.locator = locator
+        super().__init__(parent, *text, **kwargs)
+
+    def __locator__(self):
+        if self.locator:
+            return self.locator
+        return super().__locator__()
+
+    def fill(self, value):
+        active = self.active
+        if not self.disabled and ((value and not active) or (not value and active)):
```

Comment:
```
if not value or not active:
    return
self.click()
```
sounds better

Reply:
in this case if not value will not click (if we want to dis-activate the button), with the negative exclusion condition it will be same  as with positive one, prefer the positive as more understandable at first hand.

Reply:
okay, then:
```
if value is None or not active:
    return
```

Reply:
for you example to work as expected we have to write: 
```python
if (value and active) or (not value and not active):
    return 
```

Reply:
you are talking about absolutely same logic that is used for checkbox:
```
    def fill(self, value):
        value = bool(value)
        current_value = self.active
        if value == current_value:
            return False
        else:
            self.click()
            return True
```